### ISSUE TITLE
Can't get full markup from "renderToStringWithData"

### ISSUE THREAD

**Issue by novus57** [apollographql/react-apollo#328]
I can't get full page markup, only part without queries. Intial state come with data but markup doesn't full. Sorry for my English. Thanks a lot for help.
```js
app.use((req, res) => {
  const memoryHistory = createMemoryHistory(req.url);
  const user = Object.assign({}, { current: req.user });
  if (user.current) {
    delete user.current.password;
  }

  const apolloClient = new ApolloClient({
    ssrMode: true,
    // Remember that this is the interface the SSR server will use to connect to the
    // API server, so we need to ensure it isn't firewalled, etc
    networkInterface: createNetworkInterface({
      uri: 'http://localhost:3000/graphql',
      credentials: 'same-origin',
      // transfer request headers to networkInterface so that they're
      // accessible to proxy server
      // Addresses this issue: https://github.com/matthew-andrews/isomorphic-fetch/issues/83
      headers: req.headers,
    }),
  });

  const initialState = { user };
  const store = configureStore(memoryHistory, initialState, apolloClient);
  const history = syncHistoryWithStore(memoryHistory, store);
  match(
    { history, routes: createRoutes(store), location: req.originalUrl },
    (error, redirectLocation, renderProps) => {
      if (error) {
        res.status(500).send(error.message);
      } else if (redirectLocation) {
        res.redirect(302, redirectLocation.pathname + redirectLocation.search);
      } else if (renderProps) {
        const component = (
          <ApolloProvider store={store} client={apolloClient}>
            <MuiThemeProvider muiTheme={getMuiTheme(null, { userAgent: req.headers['user-agent'] })}>
              <RouterContext {...renderProps} />
            </MuiThemeProvider>
          </ApolloProvider>
        );

        renderToStringWithData(component).then((content) => {
          res.send(renderInitialView(content.markup, content.initialState));
        }).catch(dataError => winston.error('RENDERING ERROR:', dataError));
      }
    }
  );
});
```

**Comment by tmeasday** [comment]
Hi @novus57, which version of RA are you using?

**Comment by amannn** [comment]
Is it possible that server side rendering got broken in a recent version? [GitHunt](http://www.githunt.com/) which uses [react-apollo ^1.0.0-rc.3](https://github.com/apollographql/GitHunt-React/blob/master/package.json#L57) also renders the front page only on the client side, despite having a call to [renderToStringWithData on the server side](https://github.com/apollographql/GitHunt-React/blob/master/ui/server.js#L59).

Maybe it has something to do with the `cache-and-network` `fetchPolicy`? I just ran into this too after an update of apollo-client and react-apollo with a view that uses the same `fetchPolicy`. I've now reverted everything back to apollo-client@0.10.1 and react-apollo@0.13.2 so I can't easily test if it's the fault of the `fetchPolicy`.

**Comment by helfer** [comment]
@amann that is certainly possible. I'm not sure how we would go about testing it without knowing which part might have been broken (renderToStringWithData, rehydration, not fetching on the client etc.). If anyone else is seeing this, a detailed report or reproduction would be appreciated!

**Comment by amannn** [comment]
I've just upgraded my app again to apollo 1.x. When using `fetchPolicy: 'network-only'` it works fine, but when using `fetchPolicy: 'cache-and-network',` the page renders without data. Maybe the bug is somehwere around the [`fetchData`](https://github.com/apollographql/react-apollo/blob/master/src/graphql.tsx#L433) method of the `graphql` HOC?

**Comment by jbaxleyiii** [comment]
@amannn would you be able to create a reproduction test of this?

**Comment by jbaxleyiii** [comment]
closing since this has been open longer than two weeks without a reproduction. Feel free to reopen if its still an issue!

**Comment by amannn** [comment]
@jbaxleyiii I've added a PR that (hopefully) fixes the issue: #688
